src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbVacancyRepository.cs

```diff
@@ -80,22 +81,20 @@ public MongoDbVacancyRepository(ILogger<MongoDbVacancyRepository> logger, IOptio
                          builder.Ne(v => v.EmployerLocation.Latitude, null) &
                          builder.Ne(v => v.EmployerLocation.Longitude, null);
 
-            var options = new FindOptions<Vacancy>
-            {
-                Sort = Builders<Vacancy>.Sort.Descending(v => v.VacancyReference),
-                Limit = 1
-            };
-
             var collection = GetCollection<Vacancy>();
-            var result = await RetryPolicy.ExecuteAsync(() => collection.FindAsync(filter, options));
-            return result.SingleOrDefault();
+            var result = await RetryPolicy.ExecuteAsync(context => collection.FindAsync(filter), new Context(nameof(GetSingleVacancyForPostcode)));
+            
+            return result
+                        .ToList()
+                        .OrderByDescending(x => x.VacancyReference)
+                        .SingleOrDefault();
```

Comment:
This needs to be FirstOrDefault as we now have many vacancies returning. Unless we reintroduce FindOptions Limit = 1.

Reply:
good spot!